safe_cli/operators/hw_accounts/hw_account_manager.py

```diff
@@ -0,0 +1,140 @@
+from enum import Enum
+from typing import Dict, List, Optional, Set, Tuple
+
+from eth_typing import ChecksumAddress
+from prompt_toolkit import HTML, print_formatted_text
+
+from gnosis.eth.eip712 import eip712_encode
+from gnosis.safe import SafeTx
+
+from safe_cli.operators.hw_accounts.hw_account import HwAccount
+
+
+class HwWalletType(Enum):
+    TREZOR = "Trezor"
+    LEDGER = "Ledger"
+
+
+class HwAccountManager:
+    def __new__(cls):
+        if not hasattr(cls, "instance"):
+            cls.instance = super(HwAccountManager, cls).__new__(cls)
+        return cls.instance
+
+    def __init__(self):
+        self.accounts: Set[HwAccount] = set()
+        self.supported_hw_wallets: Dict[str, HwAccount] = {}
+        try:
+            from safe_cli.operators.hw_accounts.ledger_manager import LedgerManager
+
+            self.supported_hw_wallets[HwWalletType.LEDGER] = LedgerManager
+        except (ModuleNotFoundError, IOError):
```

Comment:
I would prefer to use `ImportError` as it should cover all cases instead of ModuleNotFoundError

Also I think we should handle separately an IOError if the library is installed but Ledger cannot be connected or if the dependency is not installed, same with Trezor. If not it's confusing where the error is coming from